arwen/host/mockSmartContractCallerTest.go

```diff
@@ -0,0 +1,118 @@
+package host
+
+import (
+	"testing"
+
+	mock "github.com/ElrondNetwork/arwen-wasm-vm/mock/context"
+	worldmock "github.com/ElrondNetwork/arwen-wasm-vm/mock/world"
+	"github.com/ElrondNetwork/elrond-go/core/vmcommon"
+)
+
+type testSmartContract struct {
+	address []byte
+	balance int64
+	config  interface{}
+}
+
+type testTemplateConfig struct {
+	t        *testing.T
+	input    *vmcommon.ContractCallInput
+	useMocks bool
+}
+
+type mockTestSmartContract struct {
+	testSmartContract
+	initMethods []func(*mock.InstanceMock, interface{})
+}
+
+func createMockContract(address []byte) *mockTestSmartContract {
+	return &mockTestSmartContract{
+		testSmartContract: testSmartContract{
+			address: address,
+		},
+	}
+}
+
+func (mockSC *mockTestSmartContract) withBalance(balance int64) *mockTestSmartContract {
+	mockSC.balance = balance
+	return mockSC
+}
+
+func (mockSC *mockTestSmartContract) withConfig(config interface{}) *mockTestSmartContract {
+	mockSC.config = config
+	return mockSC
+}
+
+func (mockSC *mockTestSmartContract) withMethods(initMethods ...func(*mock.InstanceMock, interface{})) mockTestSmartContract {
+	mockSC.initMethods = initMethods
+	return *mockSC
+}
+
+func (mockSC *mockTestSmartContract) initialize(t testing.TB, host *vmHost, imb *mock.InstanceBuilderMock) {
+	instance := imb.CreateAndStoreInstanceMock(t, host, mockSC.address, mockSC.balance)
+	for _, initMethod := range mockSC.initMethods {
+		initMethod(instance, mockSC.config)
+	}
+}
+
+type mockInstancesTestTemplate struct {
+	testTemplateConfig
+	contracts     *[]mockTestSmartContract
+	setup         func(*vmHost, *worldmock.MockWorld)
+	assertResults func(*worldmock.MockWorld, *VMOutputVerifier)
+}
+
+func runMockInstanceCallerTestBuilder(t *testing.T) *mockInstancesTestTemplate {
```

Comment:
Suggest to rename this constructor to `buildMockInstanceCallTest()`. Then rename `runTestWithMockMockInstances` to `runTest()` and make it a method of `mockInstancesTestTemplate`.

Reply:
I'll rename all thee of them: for mock call, instance call and create instance

Reply:
done